src/Plugin/Commerce/PaymentGateway/RazorpayCheckout.php

```diff
@@ -124,26 +125,61 @@ public function validateConfigurationForm(array &$form, FormStateInterface $form
 
         if ($form_state->getErrors())
         {
+            $validationErrorProperties = $this->triggerValidationInstrumentation(
+                ['error_message' => 'Key Id and or Key Secret is null'], $key_id);
+
+            \Drupal::logger('error')->error('Key Id and Key Secret are required.');
             return;
         }
 
         $form_state->setValue('id', 'razorpay');
 
         $values = $form_state->getValue($form['#parents']);
 
-        if (empty($values['key_id']) || empty($values['key_secret']))
+        if (empty($values['key_id']) or empty($values['key_secret']))
         {
             return;
         }
 
         if (substr($values['key_id'], 0, 8) !== 'rzp_' . $values['mode'])
         {
+            $validationErrorProperties = $this->triggerValidationInstrumentation(
+                ['error_message' => 'Invalid Key Id or Key Secret'], $key_id, $key_secret);
             $this->messenger()->addError($this->t('Invalid Key ID or Key Secret for ' . $values['mode'] . ' mode.'));
             $form_state->setError($form['mode']);
 
             return;
         }
 
+        $status = $form_state->cleanValues()->getValues($form['#parents'])['status'];
+
+        $query = \Drupal::database()->query("SELECT CAST(`data` AS CHAR(10000) CHARACTER SET utf8) AS decoded_data  FROM config WHERE `name` = :namekey", [':namekey' => 'commerce_payment.commerce_payment_gateway.razorpay']);
+
+        $data = unserialize($query->fetchField());
+
+        $authEvent = '';
+
+        $authProperties = [
+            'is_key_id_populated'     => true,
+            'is_key_secret_populated' => true,
+            'page_url'                => $_SERVER['REQUEST_SCHEME'] . '://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'],
+            'auth_successful_status'  => true,
+            'is_plugin_activated'     => $status
+        ];
+
+        if(empty($data['configuration']['key_id']) and empty($data['configuration']['key_secret']))
```

Comment:
@garimaprachi leave whitespace after if , every condition should start with new line